packages/react/src/component-tests/IcCheckbox/IcCheckbox.cy.tsx

```diff
@@ -127,6 +130,57 @@ describe("IcCheckbox", () => {
     cy.get('button[type="reset"]').click();
     cy.get(CHECKBOX_SELECTOR).eq(0).should(HAVE_PROP, "checked", false);
   });
+
+  it("renders checkboxes dynamically in a group", () => {
+    mount(<DynamicLoading />);
+
+    cy.get("ic-button").click();
+    cy.findShadowEl(CHECKBOX_SELECTOR, "input").should(
+      HAVE_ATTR,
+      "id",
+      "ic-checkbox-Item-1-What-is-your-favourite-coffee?"
+    );
+  });
+
+  it("passes down the name and label to child checkboxes when they update", () => {
+    mount(
+      <IcCheckboxGroup
+        label="What is your favourite coffee?"
+        name="radio-group-1"
+      >
+        <IcCheckbox value="Item 1" label="Item 1" />
+        <IcCheckbox value="Item 2" label="Item 2" name="Item 2 name" />
+      </IcCheckboxGroup>
+    );
+
+    cy.get(CHECKBOX_SELECTOR).each((item, index) => {
+      const wrappedItem = cy.wrap(item);
+      wrappedItem.should(
+        HAVE_PROP,
+        "groupLabel",
+        "What is your favourite coffee?"
+      );
+      wrappedItem.should(
+        HAVE_PROP,
+        "name",
+        index === 0 ? "radio-group-1" : "Item 2 name"
```

Comment:
```suggestion
        index === 0 ? "checkbox-group-1" : "Item 2 name"
```
Should this be checkbox-group-1?